R/c14_date_list_calibrate.R

```diff
@@ -49,44 +39,16 @@ calibrate.c14_date_list <- function(x) {
     char = "+"
   )
 
-  # add or empty columns calage and calstd
-  if (c("calage", "calstd") %in% colnames(x) %>% all) {
-    x$calage <- NA
-    x$calstd <- NA
-  } else {
-    x <- x %>%
-      tibble::add_column(
-        calage = NA,
-        .after = "c14std"
-      ) %>%
-      tibble::add_column(
-        calstd = NA,
-        .after = "calage"
-      )
-  }
-
-  # determine dates that are out of the range of the calcurve can not be calibrated
-  toona <- which(is.na(x$c14age) | is.na(x$c14std))
-  toosmall <- which(x$c14age < 100)
-  toobig <- which(x$c14age > 45000)
-  outofrange <- c(toona, toosmall, toobig) %>% unique
-
-  # copy these dates without calibration
-  x$calage[outofrange] <- x$c14age[outofrange]
-  x$calstd[outofrange] <- x$c14std[outofrange]
-
-  # 2sigma range probability threshold
-  threshold <- (1 - 0.9545) / 2
+  # add empty columns "calage" and "calprobdistr"
+  x %<>% add_or_replace_column_in_df("calage",  NA_integer_, .after = "c14std")
+  x %<>% add_or_replace_column_in_df("calprobdistr",  I(replicate(nrow(x), data.frame())), .after = "calage")
```

Comment:
calprobdistr should be optional in my opinion, because it really bloats the result object. What do you think about a switch to turn calprobdistr and calrange on and off? Default could be calprobdistr = FALSE, calrange = TRUE.